packages/components/src/box-control/index.tsx

```diff
@@ -155,57 +154,51 @@ function BoxControl( {
 	};
 
 	return (
-		<Root id={ id } role="group" aria-labelledby={ headingId }>
-			<Header className="component-box-control__header">
-				<FlexItem>
-					<BaseControl.VisualLabel id={ headingId }>
-						{ label }
-					</BaseControl.VisualLabel>
-				</FlexItem>
-				{ allowReset && (
-					<FlexItem>
-						<Button
-							className="component-box-control__reset-button"
-							variant="secondary"
-							isSmall
-							onClick={ handleOnReset }
-							disabled={ ! isDirty }
-						>
-							{ __( 'Reset' ) }
-						</Button>
-					</FlexItem>
-				) }
-			</Header>
-			<HeaderControlWrapper className="component-box-control__header-control-wrapper">
-				<FlexItem>
-					<BoxControlIcon side={ side } sides={ sides } />
-				</FlexItem>
-				{ isLinked && (
-					<FlexBlock>
-						<AllInputControl
-							aria-label={ label }
-							{ ...inputControlProps }
-						/>
-					</FlexBlock>
-				) }
-				{ ! isLinked && splitOnAxis && (
-					<FlexBlock>
-						<AxialInputControls { ...inputControlProps } />
-					</FlexBlock>
-				) }
-				{ ! hasOneSide && (
-					<FlexItem>
-						<LinkedButton
-							onClick={ toggleLinked }
-							isLinked={ isLinked }
-						/>
-					</FlexItem>
-				) }
-			</HeaderControlWrapper>
+		<Grid
```

Comment:
I moved from HStack/VStack to Grid because the semantics were incorrect. When tabbing through, the LinkedButton had focus first rather than the input. 

Moving the components in the expected order and adjusting for visual changes using grid seemed like the best approach here.

Reply:
>Using flex or grid in theory shouldn't influence the semantics (we're still using the same role and aria-labelledby attributes)

Yes, it's a bit odd! There were many layers of Flex/Flex Item/Flex Block before so it's likely that simplifying to HStack/VStack made a bit of an impact there. I took a quick look to see if there was a different order or reversing set previously, but I didn't see anything. So I figured it was worth switching to Grid for the flexibility rather than digging into how it was in a different order before, but the tab order was correct. 

>why does the grid have 3 columns of equal width (1fr) ?
why does each "row" span for 4 columns, if the grid should have only 3 ?
why is the link/unlink button assigned to the 4th column, if the grid should have only 3 columns?
why is the reset button not explicitly assigned to a grid cell? Relying on the grid algorithm to automatically place it could easily introduce unexpected changes in the future

It was a bit organic and mainly related to your last question. When changing the number of columns in the Grid column, I noticed that an additional column was generated outside of the set number, due to the span. It seemed to work well because the last column appeared to have an inline width, rather than having another equal width fourth column.

But it makes sense that it would be an unstable way to approach it, so I can more explicitly set up four columns with the various widths.

Reply:
Not sure if there is another way to achieve this without any hardcoded values:

https://github.com/WordPress/gutenberg/blob/493e8d9d7f9a26b8820b3435d943f43a511a6546/packages/components/src/box-control/index.tsx#L160

With the two values being for the buttons and the icon/input/slider taking up the remaining space.

Reply:
There is! We can use [the `fr` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_grid_layout/Basic_concepts_of_grid_layout#the_fr_unit) and the [`min-content` CSS keyword](https://developer.mozilla.org/en-US/docs/Web/CSS/min-content)

Something like 

`templateColumns="1fr min-content min-content"`

should do the trick

Reply:
Ah yes, I meant three columns. And thanks for sharing, TIL about `min-content`!